pkg/controller/compute/droplet.go

```diff
@@ -130,6 +130,27 @@ func (c *dropletExternal) Observe(ctx context.Context, mg resource.Managed) (man
 		cr.SetConditions(xpv1.Available())
 	}
 
+	if currentSpec.ProjectID != nil {
+		page := 0
+		for {
```

Comment:
so this doesn't seem specific to droplets, an improvement (especially if doing this for other resources) would be to make this a generic function of the base client so that the individual resource clients can just do something like `client.GetProject` and return nil if it doesn't exist.